File: backend/pkg/model/request/testcase.go
package request

type Testcase struct {
	ID      uint   `json:"id" form:"id" binding:"required"`
	Name    string `json:"name" form:"name" binding:"required"`
	Path    string `json:"path" form:"path" binding:"required"`
	Status  int32  `json:"status" form:"status" binding:"required"`
	Content string `json:"content" form:"content" binding:"required"`
}

type BatchUpdateTestcaseRequest struct {
	Data []*Testcase `json:"data" form:"data" binding:"required"`
}

File: frontend/src/model/testcase.ts
import { BaseResp } from "./resp";

export const TestcaseStatusActive = 1;
export const TestcaseStatusInactive = 2;

export interface Testcase {
    id?: number;
    name: string;
    status: number;
}

export interface BatchUpdateTestcaseRequest {
    data: Testcase[];
}

export interface BatchUpdateTestcaseResponse extends BaseResp { }
